controllers/image_generator.go

```diff
@@ -0,0 +1,407 @@
+	} else {
+		provider = r.getCloudProviderFromInfra()
+	}
+
+	switch provider {
+	case "aws":
+		r.CMimageIDKey = peerpodsCMAWSImageKey
+	case "azure":
+		r.CMimageIDKey = peerpodsCMAzureImageKey
+	default:
+		return fmt.Errorf("getCloudProvider: Unsupported cloud provider: %s", provider)
+	}
+	r.provider = provider
+
+	igLogger.Info("Cloud provider fetched successfully", "provider", r.provider, "keyID", r.CMimageIDKey)
+	return nil
+}
+
+func (r *ImageGenerator) createJobFromFile(jobFileName string) (*batchv1.Job, error) {
+	igLogger.Info("Create Job out of YAML file", "jobFileName", jobFileName)
+	yamlData, err := readJobYAML(jobFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	job, err := parseJobYAML(yamlData)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := r.client.Create(context.TODO(), job); err != nil {
+		if k8serrors.IsAlreadyExists(err) {
+			igLogger.Info("Image Job already exists, get existing object", "jobFileName", jobFileName)
+			latest := &batchv1.Job{}
+			if err := r.client.Get(context.TODO(), types.NamespacedName{
+				Name:      job.Name,
+				Namespace: job.Namespace,
+			}, latest); err != nil {
+				igLogger.Info("Image Job already exists, getting existing object failed", "jobFileName", jobFileName)
+				return nil, err
+			}
+			return latest, nil
+		} else {
+			return nil, err
+		}
+	}
+	igLogger.Info("Job file has been processed successfully and Job has been created")
+	return job, nil
+}
+
+func (r *ImageGenerator) deleteJobFromFile(jobFileName string, keepPods bool) error {
+	igLogger.Info("delete job from YAML file", "jobFileName", jobFileName)
+	yamlData, err := readJobYAML(jobFileName)
+	if err != nil {
+		return err
+	}
+
+	job, err := parseJobYAML(yamlData)
+	if err != nil {
+		return err
+	}
+
+	var deletePolicy metav1.DeletionPropagation
+	if keepPods {
+		deletePolicy = metav1.DeletePropagationOrphan // this leaves the job's pods around for tracking
+	} else {
+		deletePolicy = metav1.DeletePropagationBackground // this deletes the job's pods as well
+	}
+
+	if err := r.client.Delete(context.TODO(), job, &client.DeleteOptions{PropagationPolicy: &deletePolicy}); err != nil {
+		if k8serrors.IsNotFound(err) {
+			igLogger.Info("Image Job doesn't exist, nothing to delete", "jobFileName", jobFileName)
+		} else {
+			return err
+		}
+	}
+
+	igLogger.Info("Job file has been processed successfully and Job has been deleted", "keepPods", keepPods)
+	return nil
+}
+
+func (r *ImageGenerator) getPeerPodsCM() (*corev1.ConfigMap, error) {
+	igLogger.Info("Getting peer-pods ConfigMap")
+	peerPodsCM := &corev1.ConfigMap{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      peerpodsCMName,
+		Namespace: "openshift-sandboxed-containers-operator",
+	}, peerPodsCM)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: check in peer-pods-secret as well?
+
+	return peerPodsCM, nil
+}
+
+func (r *ImageGenerator) imageJobRunner(op string) (bool, ctrl.Result) {
```

Comment:
What are the exact semantics of the `bool` return value ? It seems to be basically `false` when some error occured and `true` otherwise... why not returning the error instead ?

Reply:
The variable name used in the controller.go implies it but maybe i should have add a comment..
The bool says whether the image creation job is completed/needed and can continue **or** it still running and should return the ctrl.Result to break the loop so it will be called again in future iteration